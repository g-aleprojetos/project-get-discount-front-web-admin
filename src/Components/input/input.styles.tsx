import styled from 'styled-components';
import colors from 'resources/colors';
import {TextH4} from 'Components/text';

export type PropsInput = React.InputHTMLAttributes<HTMLInputElement>;

export const Container = styled.div`
  display: flex;
  width: 100%;
`;

export const ContainerInput = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

export const ContainerInputBox = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  border-radius: 2px;
  border: 1px solid ${colors.shuttleGray};
`;

export const LabelBox = styled.label`
  display: flex;
  flex-direction: column;
  gap: 4px;
`;

export const Input = styled.input`
  display: flex;
  height: 30px;
  width: 100%;
  padding: 8px 0;
  text-indent: 8px;
  font-size: 16px;
  border: none;
  background-color: ${colors.white};
  color: ${colors.black};
  outline: 0;
`;

export const MostrarSenha = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0 10px;
  height: 30px;
  cursor: pointer;
`;

export const TextoLabel = styled(TextH4).attrs({
  cursor: 'pointer',
})``;
