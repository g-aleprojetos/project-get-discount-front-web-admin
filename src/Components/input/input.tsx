import React, {HTMLInputTypeAttribute, forwardRef, useState} from 'react';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faEye, faEyeSlash} from '@fortawesome/free-solid-svg-icons';

import * as S from './input.styles';
import colors from 'resources/colors';

type EyeIconProps = {
  isVisible: boolean;
  toggleVisibility: () => void;
};

type Props = S.PropsInput & {
  testId?: string;
  label?: string;
  corlabel?: string;
  value?: string;
  tipo?: HTMLInputTypeAttribute;
  requered?: boolean;
  password?: boolean;
};

export const Input = forwardRef<HTMLInputElement, Props>((props, ref) => {
  const {testId, label, tipo = 'text', corlabel = colors.blue} = props;

  const [mostrarSenha, setMostrarSenha] = useState(false);
  const tipoPassword = tipo === 'password';
  const tipoInput = tipoPassword ? (mostrarSenha ? 'text' : 'password') : tipo;

  const togglePasswordVisivel = () => {
    setMostrarSenha(!mostrarSenha);
  };

  return (
    <S.Container>
      <S.ContainerInput data-testid={testId ?? 'input-pardao'}>
        <S.LabelBox data-testid={testId ? testId + '-label' : 'input-label'}>
          <S.TextoLabel cor={corlabel}>{label}</S.TextoLabel>
          <S.ContainerInputBox>
            <S.Input
              data-testid="input-padrao-box"
              type={tipoInput}
              {...props}
              ref={ref}
            />
            {tipoPassword && (
              <S.MostrarSenha>
                <EyeIcon
                  isVisible={mostrarSenha}
                  toggleVisibility={togglePasswordVisivel}
                />
              </S.MostrarSenha>
            )}
          </S.ContainerInputBox>
        </S.LabelBox>
      </S.ContainerInput>
    </S.Container>
  );
});

Input.displayName = 'Input';

const EyeIcon: React.FC<EyeIconProps> = ({isVisible, toggleVisibility}) => {
  return (
    <FontAwesomeIcon
      data-testid={isVisible ? 'icone-olho-aberto' : 'icone-olho-fechado'}
      icon={isVisible ? faEye : faEyeSlash}
      onClick={toggleVisibility}
    />
  );
};
