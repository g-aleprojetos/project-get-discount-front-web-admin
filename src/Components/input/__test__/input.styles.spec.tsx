import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from '../input.styles';

describe('Input.styles', () => {
  test('Container DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Container />).toJSON();

    expect(container).toMatchSnapshot();
  });
  test('ContainerInput DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.ContainerInput />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerInputBox DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.ContainerInputBox />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('LabelBox DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.LabelBox />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('InputBox DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Input />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('MostrarSenha DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.MostrarSenha />).toJSON();

    expect(container).toMatchSnapshot();
  });
});
