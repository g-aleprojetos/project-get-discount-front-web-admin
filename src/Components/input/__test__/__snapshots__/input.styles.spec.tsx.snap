// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.styles Container DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

<div
  className="c0"
/>
`;

exports[`Input.styles ContainerInput DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

<div
  className="c0"
/>
`;

exports[`Input.styles ContainerInputBox DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  border-radius: 2px;
  border: 1px solid #545d6A;
}

<div
  className="c0"
/>
`;

exports[`Input.styles InputBox DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 30px;
  width: 100%;
  padding: 8px 0;
  text-indent: 8px;
  font-size: 16px;
  border: none;
  background-color: #FFFFFF;
  color: #000000;
  outline: 0;
}

<input
  className="c0"
/>
`;

exports[`Input.styles LabelBox DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
}

<label
  className="c0"
/>
`;

exports[`Input.styles MostrarSenha DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 10px;
  width: 30px;
  height: 30px;
  cursor: pointer;
}

<div
  className="c0"
/>
`;
